src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts

```diff
@@ -88,6 +96,13 @@ export class ServicesComponent extends ListWithDetails implements OnChanges, OnI
     ];
   }
 
+  openModal() {
+    this.router.url.includes('/services')
+      ? this.router.navigate([BASE_URL, { outlets: { modal: [URLVerbs.CREATE] } }])
+      : (this.bsModalRef = this.modalService.show(ServiceFormComponent)),
+      (this.bsModalRef.componentInstance.hiddenServices = this.hiddenServices);
```

Comment:
The ternary operator is nice for small assignment expressions:
```ts
value = condition ? this : that
```
But in this case it's more readable to write the if-else block.

That said, why is this conditional to the URL? This basically means the component stills alters its behaviour based on an external concern, rather than on an explicit input argument.

I'd do the same thing: let's make the 'modal/modeless' an input argument, and force the Wizard to request this component to be modal.

Reply:
Thanks @epuertat . Updated with your suggestion.

Reply:
>  why is this conditional to the URL?

@epuertat The condition is there because using a routed modal in the cluster expansion causes issues when going backward steps. So I came to the conclusion that its not good to use a routed modal in cluster expansion so I went with a normal modal, but in the normal flow it should open up the routed modal.

Reply:
thanks @nizamial09 ! I'm not sure if I got all the depth of the issue though 😅

Reply:
I'll try to elaborate a bit. Let's say on the first step, I click on the `Add` button and it opens up the `Add Host` form in a routed modal. Then I click cancel and I goto the second step. Then If I click back it'll come back to the previous step (that is the Add Host step) and right away it'll open the `Add Host` form because it was routed. It's got something to do with the auxilary routes and I tried to get rid of it but no solution couldn't solve this issue. That's why I concluded that using a routed modal for forms in cluster expansion section is not a good idea. Instead I started using a regular modal here. @epuertat